File: C++ solutions/402. Remove K Digits.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        string ans="";
        for(char &ch:num)
        {
            while(ans.size() && ans.back()>ch && k)
            {
                ans.pop_back();
                k--;
            }
            if(ans.size() || ch!='0') ans.push_back(ch);
        }
        while(ans.size() && k--) ans.pop_back();
        
        return (ans=="") ? "0" : ans;
    }
};

File: JavaScript solutions/402. Remove K Digits.js
/**
 * @param {string} num
 * @param {number} k
 * @return {string}
 */
var removeKdigits = function (num, k) {
  let ans = "";

  for (let ch of num) {
    while (ans.length && ans[ans.length - 1] > ch && k) {
      ans = ans.slice(0, -1);
      k--;
    }
    if (ans.length || ch !== "0") {
      ans += ch;
    }
  }

  while (ans.length && k--) {
    ans = ans.slice(0, -1);
  }

  return ans === "" ? "0" : ans;
};
